File: app/src/main/java/vergecurrency/vergewallet/viewmodel/ShowPaperkeyViewModel.java
package vergecurrency.vergewallet.viewmodel;

import androidx.lifecycle.ViewModel;

import vergecurrency.vergewallet.service.model.MnemonicManager;

import static vergecurrency.vergewallet.service.model.PreferencesManager.getMnemonic;

public class ShowPaperkeyViewModel extends ViewModel {
	String[] seed;

	public ShowPaperkeyViewModel() {
		seed = MnemonicManager.getMnemonicFromJSON(getMnemonic());

	}


	public String[] getSeed() {
		return seed;
	}

	public void setSeed(String[] seed) {
		this.seed = seed;
	}

	public String getSeedAsText() {
		StringBuilder result = new StringBuilder();
		for (String s:seed) {
			result.append(s).append(" ");
		}
		return result.toString();
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ShowPaperkeyViewModel.kt
package vergecurrency.vergewallet.viewmodel

import androidx.lifecycle.ViewModel
import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager.Companion.mnemonic
import vergecurrency.vergewallet.service.model.MnemonicManager

class ShowPaperkeyViewModel : ViewModel() {
    var seed: Array<ByteArray>

    val seedAsText: String
        get() {
            val result = StringBuilder()
            for (s in seed) {
                result.append(String(s)).append(" ")
            }
            return String(result)
        }

    init {
        seed = MnemonicManager.getMnemonicFromJSON(mnemonic!!)!!

    }

}
